sky/optimizer.py

```diff
@@ -260,8 +260,17 @@ def _estimate_nodes_cost_or_time(
             num_resources = len(node.get_resources())
             for orig_resources, launchable_list in launchable_resources.items():
                 if not launchable_list:
+                    specified_resources_str = ''
+                    if node.get_resources():
+                        specified_resources = list(node.get_resources())[0]
+                        specified_resources_str = f' ({specified_resources})'
+                        if specified_resources.region:
+                            specified_resources_str = (
+                                f' ({specified_resources} '
+                                f'in {specified_resources.region})')
                     error_msg = (
-                        f'No launchable resource found for task {node}. '
+                        'Optimizer: No launchable resource found for task '
+                        f'{node}{specified_resources_str}. '
```

Comment:
Any reason to not use `Resources.__repr__()`? Is it because it shows too much or doesn't show region names?

Reply:
We are using `Resources.__repr__()` as the first part, but it does not show region, so we have to add the region info manually.